internal/service/s3/bucket_replication_configuration.go

```diff
@@ -309,6 +314,7 @@ func resourceBucketReplicationConfigurationCreate(d *schema.ResourceData, meta i
 	input := &s3.PutBucketReplicationInput{
 		Bucket:                   aws.String(bucket),
 		ReplicationConfiguration: rc,
+		Token:                    aws.String(d.Get("token").(string)),
```

Comment:
Because the field is optional, we should only send it the API request if it's available so let's move this line outside of the input struct and add lines similar to:

https://github.com/hashicorp/terraform-provider-aws/blob/19469514cc16736d80ab976aa3178d5aaedbeba8/internal/service/s3/bucket_object_lock_configuration.go#L114-L117

Reply:
Good catch!